src/action/list_directory.rs

```diff
@@ -56,24 +56,23 @@ impl From<Error> for session::Error {
     }
 }
 
-struct UnsupportedValueMessage {
+#[derive(Debug)]
+struct UnsupportedValueError {
     field: String,
     value: String,
 }
 
-impl std::fmt::Debug for UnsupportedValueMessage {
-
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        f.debug_struct("UnsupportedValueMessage")
-            .field("field", &self.field)
-            .field("value", &self.value)
-            .finish()
-    }
+fn unsupported_value(field: String, value: String) -> session::ParseError {
+    session::ParseError::malformed(ParseError::UnsupportedValue(
+        UnsupportedValueError {
+            field,
+            value,
+        }))
```

Comment:
This should be a method of `ParseError`. Also, it is missing documentation.

Reply:
Fixed.